include/z64animation.h

```diff
@@ -239,95 +239,95 @@ s16 Animation_GetLength(void* animation);
 s16 Animation_GetLastFrame(void* animation);
 
 /*
- * Animation requests
+ * Animation Task Queue
  */
 
 typedef enum {
-    /* 0 */ ANIMENTRY_LOADFRAME,
-    /* 1 */ ANIMENTRY_COPYALL,
-    /* 2 */ ANIMENTRY_INTERP,
-    /* 3 */ ANIMENTRY_COPYTRUE,
-    /* 4 */ ANIMENTRY_COPYFALSE,
-    /* 5 */ ANIMENTRY_MOVEACTOR
-} AnimationType;
+    /* 0 */ ANIMTASK_LOAD_PLAYER_FRAME,
+    /* 1 */ ANIMTASK_COPY,
+    /* 2 */ ANIMTASK_INTERP,
+    /* 3 */ ANIMTASK_COPY_USING_MAP,
+    /* 4 */ ANIMTASK_COPY_USING_MAP_INVERTED,
+    /* 5 */ ANIMTASK_ACTOR_MOVE
+} AnimTaskType;
 
 typedef struct {
-    /* 0x000 */ DmaRequest req;
-    /* 0x020 */ OSMesgQueue msgQueue;
-    /* 0x038 */ OSMesg msg;
-} AnimEntryLoadFrame; // size = 0x3C
+    /* 0x00 */ DmaRequest req;
+    /* 0x20 */ OSMesgQueue msgQueue;
+    /* 0x38 */ OSMesg msg;
+} AnimTaskLoadPlayerFrame; // size = 0x3C
 
 typedef struct {
-    /* 0x000 */ u8 queueFlag;
-    /* 0x001 */ u8 vecCount;
-    /* 0x004 */ Vec3s* dst;
-    /* 0x008 */ Vec3s* src;
-} AnimEntryCopyAll; // size = 0xC
+    /* 0x00 */ u8 group;
```

Comment:
`group` makes it sound like this is an id, but I was surprised to find that this is a bitset. Maybe `groups` or `groupFlags`?

Reply:
not sure if this became clear as you read more of the code, but its always going to be just one group number. it is handled with bits yes, but this value within each task will never hold multiple groups/bit numbers. it is a single group number that increments as "1, 2, 4, 8..." instead of "1, 2, 3..."

Reply:
you can think of them as "regular" incrementing number sure, but the system never operates on it that way. The current group number is initialized with 1 (bit position 0) and incremented with `<<= 1`.
I would feel different if there was some interface with which a user can pass a group number as a "whole number" and the system does the shifting interpretation on its end. but this isnt the case.

It could be called a `groupBit` or something, but this is being more obtuse than needed imo, its still a single number that identifies a group, even if it doesnt count up normally.

Reply:
Hmm I think `sDisableAnimQueueFlags`/`sDisabledTransformTaskGroups` is important here. It doesn't really make sense to or-together the different group ids; normally the only thing you're allowed to do with ids is compare them for equality. And the `if (!(task->group & sDisabledTransformTaskGroups))` is also pretty mysterious.

But it becomes clearer if you think of `sDisabledTransformTaskGroups` as a bitset where each bit represents a group i.e. group 0 is represented by bit 0, group 1 is represented by bit 1, and so on. Then `group` and `sCurAnimTaskGroup` are also group bitsets with one element set, or `(1 << group_num)` where `group_num` is the "real" group number (although it's never actually written down anywhere), and so `if (!(task->group & sDisabledTransformTaskGroups))` is testing whether the group number is contained in the set of disabled groups.

In summary: I still think this and `sCurAnimTaskGroup` should be thought of as `(1 << group_num)`, rather than single number that identifies a group, based on how `sDisabledTransformTaskGroups` is used. I think the names and comments should hint at that. Maybe `group` and `sCurAnimTaskGroup` is fine and my issue is more with the comment wording, the comments seem to treat this bit-twiddling as some kind of arcane implementation detail but it becomes easier to explain if you treat group numbers as bit numbers rather than powers of 2.

Reply:
ignoring the id aspect thing for a moment, I don't understand what's being "grouped"

Reply:
Different tasks or "entries" are being grouped together. Then, when `AnimTaskQueue_DisableTransformTasksForGroup` is called, the "transformative tasks" only for the currently active group (at the time of calling) will be skipped for that frame.

In practice, there are only two "groups" that get used. all of Links anim tasks are in one group, and all other actors tasks are in another. This lets Link disable the transformative tasks for his stuff, without affecting others. 
The same could be done in reverse, but `AnimTaskQueue_DisableTransformTasksForGroup` is only called by player.